molecule/external-activemq/prepare.yml

```diff
@@ -0,0 +1,55 @@
+---
+- name: Default prepare
+  import_playbook: ../default/prepare.yml
+
+- name: Prepare activemq instance
+  hosts: localhost
+  gather_facts: false
+  vars:
+    broker_name: molecule-{{ lookup('env', 'BUILD_NUMBER') }}
+  tasks:
+    - name: Create EC2 security group for mq broker
+      amazon.aws.ec2_group:
+        name: "{{ broker_name }}"
+        description: Molecule external activemq EC2 test
+        rules:
+          - proto: tcp
+            from_port: 61617
+            to_port: 61617
+            cidr_ip: "{{ ansible_ec2_local_ipv4 }}/32"
+      register: ec2_group
+
+    - name: Create mq broker
+      ansible.builtin.shell: |
+        aws mq create-broker --broker-name {{ broker_name }} \
+          --deployment-mode SINGLE_INSTANCE \
+          --engine-type ACTIVEMQ \
+          --host-instance-type mq.t3.micro \
+          --engine-version 5.15.15 \
+          --users Password=alfrescoalfresco,Username=alfresco \
+          --security-groups {{ ec2_group.group_id }} \
+          --publicly-accessible \
+          --no-auto-minor-version-upgrade
+
+    - name: Wait for creation
+      ansible.builtin.shell: |
+        aws mq describe-broker --broker-id {{ broker_name }} | \
+          jq -r .BrokerState
+      register: describe_broker_output
+      until: describe_broker_output.stdout.find("RUNNING") != -1
+      retries: 20
+      delay: 60
+
+    - name: Grab activemq hostname
+      ansible.builtin.shell: |
+        aws mq describe-broker --broker-id {{ broker_name }} | \
+          jq -r '.BrokerInstances[0].Endpoints[0]' | \
+          awk -F/ '{print $3}' | \
+          sed 's/:61617//'
+      register: activemq_hostname
```

Comment:
lol I'm surprised you duplicated this task. Why not using a simple avoiding filtering previous task with jq andparsing registred json?

Reply:
![](https://media2.giphy.com/media/vX9WcCiWwUF7G/200.gif)